README
Nagios-Plugin-OverHTTP version 0.13_004
=======================================

INSTALLATION
------------

To install this module, run the following commands:

    perl Makefile.PL
    make
    make test
    make install

If you have CPANPLUS installed (`whereis cpanp`), you can use the following
command to install this module:

    cpanp i ./

This module also comes with a check_over_http Nagios plugin, which is
located in bin/ and you will want to copy that over to your Nagios
plugins directory (usually /usr/local/libexec/nagios/)

    cp bin/check_over_http /usr/local/libexec/nagios/

SUPPORT AND DOCUMENTATION
-------------------------

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Nagios::Plugin::OverHTTP

You can also look for information at:

*   [RT, CPAN's request tracker](http://rt.cpan.org/NoAuth/Bugs.html?Dist=Nagios-Plugin-OverHTTP)

*   [AnnoCPAN, Annotated CPAN documentation](http://annocpan.org/dist/Nagios-Plugin-OverHTTP)

*   [CPAN Ratings](http://cpanratings.perl.org/d/Nagios-Plugin-OverHTTP)

*   [Search CPAN](http://search.cpan.org/dist/Nagios-Plugin-OverHTTP/)


COPYRIGHT AND LICENCE
---------------------

Copyright 2009 Douglas Christopher Wilson.

This program is free software; you can redistribute it and/or
modify it under the terms of either:

*   the GNU General Public License as published by the Free
    Software Foundation; either version 1, or (at your option) any
    later version, or

*   the Artistic License version 2.0.


Changes
Revision history for Perl 5 distribution Nagios-Plugin-OverHTTP

0.15 2011-08-04
 [ENHANCEMENTS]
 - Add new Nagios::Plugin::OverHTTP::Middleware::StatusPrefix module.
 - The response attribute is now optional in Nagios::Plugin::OverHTTP::Response.

 [OTHER]
 - Change all uses of Readonly to Const::Fast.

0.14 2012-03-07
 [ENHANCEMENTS]
 - Added new feature where performance data is parsed from the body and
   X-Nagios-Performance header (in Parser::Standard).
 - If Content-Type is text/html or XHTML, then HTML::Strip is used to
   remove all (X)HTML tags (in Parser::Standard).
 - Added full support for handling performance data from the remote plugin.
   Use critical or warning command line switches to monitor performance data.

 [OTHER]
 - Fixed minor POD mistakes.

0.13 2009-10-24
 [BUG FIXES]
 - Fixed a regular expression in t/mock.t to work with buggy perl
   5.8.8 and lower.

0.12 2009-10-24
 [ENHANCEMENTS]
 - New option verb which sepcifies the HTTP verb to use when making
   the HTTP request.
 - New header supported: X-Nagios-Information which replaces using
   the body for the plugin output (message). Please refer to the
   documentation in Nagios::Plugin::OverHTTP about this header.
   Code and idea contributed by Alex Wollangk.

0.11 2009-10-04
 [BUG FIXES]
 - The check_over_http plugin how correctly works without flaws under
   ePN and also adds a line to explicity enable ePN for the plugin.

 [OTHER]
 - The check_over_http plugin now no longer automatically installs;
    see README for instructions for installing it.

0.10 2009-09-10
 [ENHANCEMENTS]
 - Add new option autocorrect_unknown_html (and enabled it by default)
   which will take a response that did not send a Nagios service status
   and is a multi line HTML message find the <title> or first <h1>
   element and add that as the first line of the message.

0.09 2009-09-10
 [ENHANCEMENTS]
 - Added new option default_status.

 [OTHER]
 - Moved around and improved documentation.

0.08 2009-06-13
 [OTHER]
 - List LWP::UserAgent as dependency in documentation.
 - Added what the defaults were to some of the usage.
 - Corrected some spelling mistakes in the documentation.
 - Changed message to "Connection refused" if the TCP connection
   cannot be established.

0.07 2009-05-20
 [ENHANCEMENTS]
 - Plugin dies if not enough information provided before a check.
 - Changed check_over_http to a default timeout of 10 seconds.
 - Allow X-Nagios-Status to be digits.

 [OTHER]
 - Added documentation for the HTTP response format.

0.06 2009-05-18
 [ENHANCEMENTS]
 - Added new option --timeout to the plugin.

 [OTHER]
 - Added `MANIFEST` author test.
 - Added documentation for the constructors.

0.05 2009-05-14
 [BUG FIXES]
 - MANIFEST was missing 2 files, and so was the distribution; previous
   version 0.04 will never successfully install.

0.04 2009-05-13
 [ENHANCEMENTS]
 - Added new option --hostname to the plugin.
 - Added new option --path to the plugin.
 - Added new option --ssl to the plugin.
 - Added parameter constraint checking.

 [OTHER]
 - Removed unused Switch from code to make perl 5.11.0 happy

0.03 2009-05-05
 [BUG FIXES]
 - Fixed issue where version was not listed as a dependency.

0.02 2009-05-01
 [BUG FIXES]
 - Fixed issue where Readonly was not listed as a dependency.

0.01 2012-04-30
 - First release.


Makefile.PL
use inc::Module::Install;

use Module::Install::AuthorRequires 0.02;

name     'Nagios-Plugin-OverHTTP';
author   'Douglas Christopher Wilson <doug@somethingdoug.com>';
license  'perl';
all_from 'lib/Nagios/Plugin/OverHTTP.pm';

# Resources
resources homepage   => 'http://github.com/dougwilson/perl5-nagios-plugin-overhttp/';
resources repository => 'git://github.com/dougwilson/perl5-nagios-plugin-overhttp.git';

# Modules that are required
requires
	'Data::Validate::Domain'    => '0.02',
	'Data::Validate::URI'       => '0.05',
	'Carp'                      => 0,
	'Const::Fast'               => 0,
	'English'                   => 0,
	'Env::Path'                 => '0.04',
	'Getopt::Long::Descriptive' => 0,
	'HTML::Strip'               => '1.04',
	'HTTP::Request'             => '5.827',
	'HTTP::Status'              => '5.817',
	'IPC::System::Simple'       => '0.13',
	'LWP::UserAgent'            => 0,
	'Moose'                     => '0.74',
	'Moose::Role'               => '0.74',
	'MooseX::Clone'             => '0.05',
	'MooseX::Getopt'            => '0.19',
	'MooseX::StrictConstructor' => '0.08',
	'MooseX::Types'             => '0.08',
	'MooseX::Types::Moose'      => 0,
	'Regexp::Common'            => '2.119',
	'Scalar::Util'              => '1.19',
	'Try::Tiny'                 => '0.04',
	'URI'                       => 0,
	'namespace::clean'          => '0.04';

# Modules that are required for tests in t/
test_requires
	'HTTP::Response'   => 0,
	'HTTP::Status'     => '5.817',
	'Test::Exception'  => 0,
	'Test::MockObject' => 0,
	'Test::More'       => '0.94';

# Modules that are required for tests in xt/
author_requires
	'Test::More'     => 0,
	'Test::Requires' => '0.02';

# Enable tests to be in multi-level directories
tests_recursive;

# Write the provided classes in the META
auto_provides;

if (eval "require Getopt::Long::Descriptive 0.082; 1;") {
	# Getopt::Long::Descriptive is install, make sure it's not a broken version
	if (Getopt::Long::Descriptive->VERSION ge '0.082'
	    && Getopt::Long::Descriptive->VERSION lt '0.084') {
		# Two versions of Getopt::Long::Descriptive have a bug that affects this
		# package, and so if you have that version, we require the version where
		# it was fixed.
		requires 'Getopt::Long::Descriptive' => '0.084';
	}
}

WriteAll;
